scipy/interpolate/tests/test_interpolate.py

```diff
@@ -2613,6 +2613,29 @@ def test_broadcastable_input(self):
                          interp_points3.size ==
                          interp_points4.size, True)
 
+    def test_read_only(self):
+        # input data
+        np.random.seed(0)
+        x = np.random.random(10)
+        y = np.random.random(10)
+        z = np.hypot(x, y)
+        
+        # x-y grid for interpolation
+        X = np.linspace(min(x), max(x))
+        Y = np.linspace(min(y), max(y))
+        X, Y = np.meshgrid(X, Y)
+
+        x.setflags(write=False)
+        y.setflags(write=False)
+        z.setflags(write=False)
+        X.setflags(write=False)
+        Y.setflags(write=False)
+
+        for interpolator in (NearestNDInterpolator, LinearNDInterpolator,
+                             CloughTocher2DInterpolator):
+            interp = interpolator(list(zip(x, y)), z)
+            interp((X, Y))
```

Comment:
Isn't this going to create a new array internally which will be writeable. I think that's the reason the test is passing even though `xi` isn't marked `const`.

Reply:
Just to be clear, by "internally" I mean the interpolator will effectively call `np.ascontiguousarray((X, Y))` which will make the new array. The same should happen for `list(zip(x, y))` above where it will be turned into a single ndarray.

So, I would say make sure you're passing contiguous ndarrays for all arguments in this test.

Reply:
Hi. thank you for reviewing and catching it. you are right, we need more const. I fixed it. PTAL.